### PR TITLE
Alex dev

### PR DESCRIPTION
**Pull request by AlexanderFengler**
Small change making the posterior predictive plot a little more flexible. 
Can probably be done more elegantly.

### CODE REVIEW

File: kabuki/analyze.py [hunk 1]
```diff
@@ -476,9 +476,24 @@ def _plot_posterior_pdf_node(bottom_node, axis, value_range=None, samples=10, bi
...
+            processed_data = data_processor(bottom_node.value.values)
+            
+        axis.hist(processed_data,
+                  density=True, 
+                  color='blue', #range=(value_range[0], value_range[-1]), 
+                  label='data',
+                  bins=bins, 
+                  histtype='step', 
+                  lw = 1.)
+        # axis.hist(bottom_node.value.values, density=True, color='r',
```
Comment (twiecki) [inline kabuki/analyze.py:494]: Can just remove this here.

Comment (AlexanderFengler) [reply kabuki/analyze.py]: ok committed that change

File: kabuki/analyze.py [hunk 2]
```diff
@@ -476,9 +476,21 @@ def _plot_posterior_pdf_node(bottom_node, axis, value_range=None, samples=10, bi
 
     # Plot data
     if len(bottom_node.value) != 0:
-        axis.hist(bottom_node.value.values, density=True, color='r',
-                  range=(value_range[0], value_range[-1]), label='data',
-                  bins=bins, histtype='step', lw=2.)
+        data_processor = kwargs.pop('data_preprocessor', None)
+        
+        if data_processor == None:
```
Comment (twiecki) [inline kabuki/analyze.py:481]:
```suggestion
        if data_processor is None:
```

File: kabuki/analyze.py [hunk 3]
```diff
@@ -476,9 +476,21 @@ def _plot_posterior_pdf_node(bottom_node, axis, value_range=None, samples=10, bi
...
+            print('No data_processor kwarg specified.\n If the plot fails, try to supply a data_processor function. This plot expects node data (coming from your original dataset DataFrame), \n to be supplied as a one dimensional array in a given range. If you supplied data as a two dimension array (reaction times and choices separately for example), \n use the data_processors to bring reshape if meaninfully (e.g. reaction times * choices if choices options are -1 and 1')
```
Comment (twiecki) [inline kabuki/analyze.py:482]: I don't think this `print` is a good idea, most people won't need to specify this, right? So they would just get confused.

File: kabuki/analyze.py [hunk 4]
```diff
@@ -476,9 +476,21 @@ def _plot_posterior_pdf_node(bottom_node, axis, value_range=None, samples=10, bi
 
     # Plot data
     if len(bottom_node.value) != 0:
-        axis.hist(bottom_node.value.values, density=True, color='r',
-                  range=(value_range[0], value_range[-1]), label='data',
-                  bins=bins, histtype='step', lw=2.)
+        data_processor = kwargs.pop('data_preprocessor', None)
```
Comment (twiecki) [inline kabuki/analyze.py:479]: Need to add this kwarg with an explanation to the doc string.

File: kabuki/analyze.py (hunk 3 above)
Comment (AlexanderFengler) [reply kabuki/analyze.py]: got rid of it

File: kabuki/analyze.py (hunk 4 above)
Comment (AlexanderFengler) [reply kabuki/analyze.py:479]: added to doc for `plot_posterior_predictive()`.

File: kabuki/analyze.py [hunk 5]
```diff
@@ -476,9 +476,21 @@ def _plot_posterior_pdf_node(bottom_node, axis, value_range=None, samples=10, bi
...
+        
+        if data_processor == None:
+            #print('No data_processor kwarg specified.\n If the plot fails, try to supply a data_processor function. This plot expects node data (coming from your original dataset DataFrame), \n to be supplied as a one dimensional array in a given range. If you supplied data as a two dimension array (reaction times and choices separately for example), \n use the data_processors to bring reshape if meaninfully (e.g. reaction times * choices if choices options are -1 and 1')
```
Comment (twiecki) [inline kabuki/analyze.py:482]: Best to not put commented code into a PR.

File: kabuki/analyze.py [hunk 6]
```diff
@@ -521,6 +533,11 @@ def plot_posterior_predictive(model, plot_func=None, required_method='pdf', colu
         plot_func : function (default=_plot_posterior_pdf_node)
             Plotting function to use for each observed node
             (see default function for an example).
+        
+        data_processor: function (default=None)
+            Inside plot_posterior_predictive the standard plotting function (histogram) assumes that your data is supplied as a 1-dimensional array (e.g. outcome variable in range (-x,x)).
```
Comment (twiecki) [inline kabuki/analyze.py:538]: Better line-breaks like above.

File: kabuki/analyze.py (hunk 5 above)
Comment (AlexanderFengler) [reply kabuki/analyze.py]: done

File: kabuki/analyze.py (hunk 6 above)
Comment (AlexanderFengler) [reply kabuki/analyze.py]: done

**Review by twiecki — APPROVED** [review]

**Comment by twiecki** [comment]
Thanks @AlexanderFengler!

**Pull request merged**